### PR TITLE
Gcw 2335 basic setup required for chat refactor messages

### PR DESCRIPTION
**Pull request by steveyken**
no specs yet

### CODE REVIEW

File: app/models/concerns/message_subscription.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    user_ids = []
+
+    # Add the following users
+    #   - Donor / Charity user - if offer/order is not cancelled
+    #   - Message sender
+    #   - Anyone who has previously replied to offer/order
+    #   - Admin users processing the offer/order
+    user_ids << obj.try(:created_by_id) unless obj.try('cancelled?')
+    user_ids << sender_id
+    user_ids << Subscription.where(:"#{klass}_id" => obj.id).pluck(:user_id)
```
Comment (Darshan-Chauhan) [inline app/models/concerns/message_subscription.rb]: New hash syntax?

Comment (Darshan-Chauhan) [inline app/models/concerns/message_subscription.rb]:
```suggestion
    user_ids << Subscription.where("#{klass}_id": obj.id).pluck(:user_id)
```

Comment (steveyken) [reply app/models/concerns/message_subscription.rb]: yeah I'm so old school ;)

**Comment by abhinav098** [comment]
merging this into my branch.

**Pull request merged**